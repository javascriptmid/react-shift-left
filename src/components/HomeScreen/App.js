import React from "react";
import Button from "@material-ui/core/Button";
import { Link } from "react-router-dom";

import "./App.css";

function App() {
  return (
    <div className="App">
      <header className="App-header">
        <p>
          Edit <code>src/App.js</code> and save to reload.
        </p>
        <Button variant="contained" color="primary">
          <Link to="/trivia">
            Learn React
          </Link>
        </Button>
      </header>
    </div>
  );
}

export default App;
