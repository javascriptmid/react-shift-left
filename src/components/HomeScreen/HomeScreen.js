import React from 'react';
import Button from '@material-ui/core/Button';
import { Link } from 'react-router-dom';

import Page from '../Page';
import { TitleLogo } from './components';
import Styles from './HomeScreen.module.css';

export default function HomeScreen() {
  return (
    <Page>
      <div className={Styles.wrapper}>
        <TitleLogo />

        <Link to="/trivia">
          <Button color="primary" variant="outlined">
            START PLAYING
          </Button>
        </Link>
      </div>
    </Page>
  );
}
