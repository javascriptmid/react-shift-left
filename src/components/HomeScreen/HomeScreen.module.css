.wrapper {
  display: flex;
  flex: 1;
  flex-direction: column;
  justify-content: space-between;
  align-items: center;
}
