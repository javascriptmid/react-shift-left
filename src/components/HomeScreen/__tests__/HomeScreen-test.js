import React from 'react';
import { Link } from 'react-router-dom';

import HomeScreen from '../HomeScreen';

import renderer from 'react-test-renderer';

describe('<HomeScreen />', () => {
  it('starts trivia', () => {
    const component = renderer.create(<HomeScreen />);
    expect(component).toMatchSnapshot();
  });
});
