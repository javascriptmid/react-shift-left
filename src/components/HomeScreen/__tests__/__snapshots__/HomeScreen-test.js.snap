// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<HomeScreen /> starts trivia 1`] = `
<div
  className="MuiContainer-root Page MuiContainer-maxWidthSm"
>
  <div
    className="Page"
    id="home-screen"
  >
    <div
      className="wrapper"
    >
      <img
        alt="App logo"
        className="TitleLogo"
        src="img/simpsons-logo.png"
      />
      <button
        onClick={[Function]}
      >
        <button
          className="MuiButtonBase-root MuiButton-root MuiButton-outlined MuiButton-outlinedPrimary"
          disabled={false}
          id="start-playing-button"
          onBlur={[Function]}
          onDragLeave={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          tabIndex={0}
          type="button"
        >
          <span
            className="MuiButton-label"
          >
            START PLAYING
          </span>
        </button>
      </button>
    </div>
  </div>
</div>
`;
